client/signup/main.jsx

```diff
@@ -314,9 +315,23 @@ class Signup extends Component {
 
 	getRecordProps() {
 		const { signupDependencies } = this.props;
+		let theme = get( signupDependencies, 'selectedDesign.theme' );
+
+		if ( ! theme && signupDependencies.themeParameter ) {
+			theme = signupDependencies.themeParameter;
+		}
+
+		const deps = this.getCurrentFlowSupportedQueryParams();
```

Comment:
Any reason why this does not execute `this.getDependenciesInQuery` directly? That'd would likely remove the need of converting the deps to snake case. Apparently, `getCurrentFlowSupportedQueryParams` only adds data about the site, but I think it's always empty at this point of the signup, so this data is probably useless?

Reply:
I'm not sure. Keep in mind that these changes are applied to all flows, not just ours. I thought that I should pass them just in case those props are passed. If their values are null (like in the with-theme flow), the props are omitted anyway.

Reply:
As for the need to convert them to snake case, I think we still need that because there are flows that define camelCase variables.

Reply:
> I thought that I should pass them just in case those props are passed

I think that, in that case, the flow ideally should define the site data as dependency. Anyway, not a blocker.